compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -194,6 +193,88 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         }
     }
 
+    /// Attempts to transmute an `OperandValue` to another `OperandValue`.
+    ///
+    /// Returns `None` for cases that can't work in that framework, such as for
+    /// `Immediate`->`Ref` that needs an `alloc` to get the location.
+    fn codegen_transmute_operand(
+        &mut self,
+        bx: &mut Bx,
+        operand: OperandRef<'tcx, Bx::Value>,
+        cast: TyAndLayout<'tcx>,
+    ) -> Option<OperandValue<Bx::Value>> {
+        // Callers already checked that the layout sizes match
+        debug_assert_eq!(operand.layout.size, cast.size);
+
+        let transmute_scalar = |bx: &mut Bx,
+                                mut imm: Bx::Value,
+                                from_scalar: abi::Scalar,
+                                to_scalar: abi::Scalar,
+                                to_bty: Bx::Type|
```

Comment:
Ah, right.  My first attempt captured `bx`, which blew up horribly with lifetimes errors, but now it might as well expand out as a method instead.  (Just changing the closure to a function I did try, but was kinda ugly because it meant repeating the `<Bx: Whatever>`.  Hopefully the method approach will work out, though.)